From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: David Llewellyn-Jones <david@flypig.co.uk>
Date: Wed, 5 Aug 2026 11:25:50 +0200
Subject: [PATCH 36/67] Add and adjust embedlite static prefs

Generate the EmbedLite APZ preference group required by EmbedLiteViewChild.

Restore the external-protocol permission gate with a secure default.
---
 modules/libpref/init/StaticPrefList.yaml      | 22 +++++++++++++++++++
 modules/libpref/moz.build                     |  1 +
 .../handling/ContentDispatchChooser.sys.mjs   | 10 +++++++++
 3 files changed, 33 insertions(+)

diff --git a/modules/libpref/init/StaticPrefList.yaml b/modules/libpref/init/StaticPrefList.yaml
index 35b62b572a53f..14c099a2a790e 100644
--- a/modules/libpref/init/StaticPrefList.yaml
+++ b/modules/libpref/init/StaticPrefList.yaml
@@ -5824,6 +5824,28 @@
 #endif
   mirror: always
 
+#ifdef MOZ_EMBEDLITE
+#---------------------------------------------------------------------------
+# Prefs starting with "embedlite."
+#---------------------------------------------------------------------------
+
+- name: embedlite.azpc.handle.viewport
+  type: bool
+  value: true
+  mirror: always
+
+- name: embedlite.azpc.handle.singletap
+  type: bool
+  value: true
+  mirror: always
+
+- name: embedlite.azpc.handle.longtap
+  type: bool
+  value: true
+  mirror: always
+
+#endif
+
 #---------------------------------------------------------------------------
 # Prefs starting with "extensions."
 #---------------------------------------------------------------------------
diff --git a/modules/libpref/moz.build b/modules/libpref/moz.build
index 4e582caaa5888..9c95619f5f747 100644
--- a/modules/libpref/moz.build
+++ b/modules/libpref/moz.build
@@ -44,6 +44,7 @@ pref_groups = [
     "docshell",
     "dom",
     "editor",
+    "embedlite",
     "extensions",
     "fission",
     "font",
diff --git a/toolkit/mozapps/handling/ContentDispatchChooser.sys.mjs b/toolkit/mozapps/handling/ContentDispatchChooser.sys.mjs
index 06451de5e72dc..4e78c445908b2 100644
--- a/toolkit/mozapps/handling/ContentDispatchChooser.sys.mjs
+++ b/toolkit/mozapps/handling/ContentDispatchChooser.sys.mjs
@@ -20,6 +20,12 @@ XPCOMUtils.defineLazyPreferenceGetter(
   "network.protocol-handler.prompt-from-external",
   true
 );
+XPCOMUtils.defineLazyPreferenceGetter(
+  gPrefs,
+  "requireExternalProtocolPermission",
+  "security.external_protocol_requires_permission",
+  true
+);
 
 const PROTOCOL_HANDLER_OPEN_PERM_KEY = "open-protocol-handler";
 const PERMISSION_KEY_DELIMITER = "^";
@@ -277,6 +283,10 @@ export class nsContentDispatchChooser {
    * @returns {boolean} - true if permission is set, false otherwise.
    */
   _hasProtocolHandlerPermission(scheme, aPrincipal, aTriggeredExternally) {
+    if (!gPrefs.requireExternalProtocolPermission) {
+      return true;
+    }
+
     // If a handler is set to open externally by default we skip the dialog.
     if (
       Services.prefs.getBoolPref(
